### DISCUSSION TITLE
Multiple brokers

### DISCUSSION THREAD

**Discussion by Sobes76rus** [taskiq-python/taskiq#161] [Ideas]
Hello there! I am having a problem when starting worker process with multiple brokers defined in the application
I wish to start only 1 broker as worker, but coz of `is_worker_process` defined as a class variable of `AsyncBroker`  all brokers marked as worker processes. And `available_tasks` also contains all the tasks from all brokers
Can we make some changes for mult-ibroker systems?

**Comment by s3rius** [comment]
Yes. We will do it. Actually this seems to me problematic.

**Reply by s3rius** [reply]
I made a PR that solves your issue. 

https://github.com/taskiq-python/taskiq/pull/169

After it's merged you'll be able to run multiple brokers without any problem.

**Reply by s3rius** [reply]
With release https://github.com/taskiq-python/taskiq/releases/tag/0.8.0 you can now have multiple brokers without any problem. Now `is_worker_process` variable is a local field of AsyncBroker, so only broker who is actual worker now will have this value set to true.
